File: TaylorCode/app/src/main/java/test/taylor/com/taylorcode/util/DimensionUtil.java
package test.taylor.com.taylorcode.util;

import android.content.Context;
import android.content.res.Resources;


public final class DimensionUtil {

    public static int dp2px(double dpValue) {
        final float scale = Resources.getSystem().getDisplayMetrics().density;
        return (int) (dpValue * scale + 0.5f);
    }

    public static int getScreenWidth(Context context) {
        return context.getResources().getDisplayMetrics().widthPixels;
    }
    public static int getScreenHeight(Context context) {
        return context.getResources().getDisplayMetrics().heightPixels;
    }
}

File: TaylorCode/app/src/main/java/test/taylor/com/taylorcode/util/DimensionUtil.kt
package test.taylor.com.taylorcode.util

import android.content.res.Resources
import android.util.TypedValue

fun Float.dp(): Int =
    TypedValue.applyDimension(
        TypedValue.COMPLEX_UNIT_DIP,
        this,
        Resources.getSystem().displayMetrics
    ).toInt()

